modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js

```diff
@@ -289,18 +290,157 @@ function add_dependency_permutations(o, deps) {
 		o.depends(res[i]);
 }
 
+function create_channel_graph(chan_analysis, freq_tbl) {
+	var is5GHz = freq_tbl[0].mhz >= 5000,
+	    columns = is5GHz ? freq_tbl.length * 4 : freq_tbl.length + 3,
+	    chan_graph = chan_analysis.graph,
+	    G = chan_graph.firstElementChild,
+	    step = (chan_graph.offsetWidth - 2) / columns,
+	    curr_offset = step;
+
+	function createGraphHLine(graph, pos) {
+		var elem = document.createElementNS('http://www.w3.org/2000/svg', 'line');
+		elem.setAttribute('x1', pos);
+		elem.setAttribute('y1', 0);
+		elem.setAttribute('x2', pos);
+		elem.setAttribute('y2', '100%');
+		elem.setAttribute('style', 'stroke:black;stroke-width:0.1');
+		graph.appendChild(elem);
+	}
+	
+	function createGraphText(graph, pos, text) {
+		var elem = document.createElementNS('http://www.w3.org/2000/svg', 'text');
+		elem.setAttribute('y', 15);
+		elem.setAttribute('style', 'fill:#eee; font-size:9pt; font-family:sans-serif; text-shadow:1px 1px 1px #000');
+		elem.setAttribute('x', pos + 5);
+		elem.appendChild(document.createTextNode(text));
+		graph.appendChild(elem);
+	}
+
+	chan_analysis.col_width = step;
+
+	createGraphHLine(G,curr_offset);
+	for (var i=0; i< freq_tbl.length;i++) {
+		var channel = freq_tbl[i].channel
+		chan_analysis.offset_tbl[channel] = curr_offset+step;
+
+		createGraphHLine(G,curr_offset+step);
+		createGraphText(G,curr_offset+step, channel);
+		curr_offset += step;
+
+		if (is5GHz && freq_tbl[i+1]) {
+			var next_channel = freq_tbl[i+1].channel;
+			/* Check if we are transitioning to another 5Ghz band range */
+			if ((next_channel - channel) == 4) {
+				for (var j=1; j < 4; j++) {
+					chan_analysis.offset_tbl[channel+j] = curr_offset+step;
+					createGraphHLine(G,curr_offset+step);
+					curr_offset += step;
+				}
+			} else {
+				chan_analysis.offset_tbl[channel+1] = curr_offset+step;
+				createGraphHLine(G,curr_offset+step);
+				curr_offset += step;
+
+				chan_analysis.offset_tbl[next_channel-2] = curr_offset+step;
+				createGraphHLine(G,curr_offset+step);
+				curr_offset += step;
+
+				chan_analysis.offset_tbl[next_channel-1] = curr_offset+step;
+				createGraphHLine(G,curr_offset+step);
+				curr_offset += step;
+			}
+		}
+	}
+	createGraphHLine(G,curr_offset+step);
+}
+
+function add_wifi_to_graph(chan_analysis, res, scanCache) {
+	if (scanCache[res.bssid].color == null)
+		scanCache[res.bssid].color = '#'+('00000'+(Math.random()*(1<<24)|0).toString(16)).slice(-6);
```

Comment:
I suggest to use the logic from [`firewall.getColorForName()`](https://github.com/openwrt/luci/blob/master/modules/luci-base/htdocs/luci-static/resources/firewall.js#L52) here, this ensures that the same network names always get the same pseudorandom colors.

Reply:
To not have duplicated code, I generalized part of the `firewall.getColorForName()` and added to prng.